File: packages/transform/src/features/protobuf/ProtobufIsEncodeTransformer.ts
import { ProtobufIsEncodeProgrammer } from "@typia/core";

import { ITransformProps } from "../../ITransformProps";
import { GenericTransformer } from "../../internal/GenericTransformer";

export namespace ProtobufIsEncodeTransformer {
  export const transform = (props: ITransformProps) =>
    GenericTransformer.scalar({
      ...props,
      method: "protobuf.isEncode",
      write: ProtobufIsEncodeProgrammer.write,
    });
}

File: packages/transform/src/features/protobuf/ProtobufIsEncodeTransformer.go
package protobuf

import (
	coreprotobuf "github.com/samchon/typia/packages/core/src/programmers/protobuf"
	internal "github.com/samchon/typia/packages/transform/src/internal"
)

var ProtobufIsEncodeTransformer = protobufIsEncodeTransformerNamespace{}

type protobufIsEncodeTransformerNamespace struct{}

func (protobufIsEncodeTransformerNamespace) Transform(call internal.ITypiaCallExpression) string {
	return coreprotobuf.ProtobufIsEncodeProgrammer.Write(call.TypeText)
}
